src/Time/Units.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE ExplicitNamespaces #-}
+{-# LANGUAGE KindSignatures     #-}
+{-# LANGUAGE TypeOperators      #-}
+
+module Time.Units
+       ( -- * Time
+         Time (..)
+
+         -- ** Time data types
+       , Second
+       , MilliSecond
+       , MicroSecond
+       , NanoSecond
+       , Minute
+       , Hour
+       , Day
+       , Week
+       , Fortnight
+
+         -- ** Units
+       , SecondUnit
+       , MilliSecondUnit
+       , MicroSecondUnit
+       , NanoSecondUnit
+       , MinuteUnit
+       , HourUnit
+       , DayUnit
+       , WeekUnit
+       , FortnightUnit
+       ) where
+
+import Data.Ratio (Rational)
+
+import Time.Rational (type (%), Rat)
+
+
+newtype Time (rat :: Rat) = Time Rational
+
+-- Units
+
+type SecondUnit      = 1 % 1
+type MilliSecondUnit = 1 % 1000
+type MicroSecondUnit = 1 % 1000000
+type NanoSecondUnit  = 1 % 1000000000
+
+type MinuteUnit      = 60 % 1
+type HourUnit        = 360 % 1
+type DayUnit         = 8640 % 1
+type WeekUnit        = 60480 % 1
```

Comment:
I think it's a good idea to add a couple extra families which allow us to specify this time units in more convenient way. Something like this:

```haskell
type WeekUnit = (5 % 1) %* Day
```

This can be achieved with the following type family:

```haskell
type family (%*) (m :: Rat) (n :: Rat) where ...
```

Not sure about `%*` name, but `*` is already taken for `Nat`...
  
Maybe for even nicer API we might want to write something like this:

```
type MilliSecondUnit = Second / 1000
type MinuteUnit = 60 :* Second
```

Which results in couple extra couple type families. What do you think?